core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -4210,6 +4247,131 @@ public boolean isExistingUser(String userName) throws UserStoreException {
         return userList;
     }
 
+    /**
+     * CountUsers in the user stores
+     *
+     * @param filter The string to filter out user
+     * @return userCount
+     * @throws UserStoreException UserStoreException
+     */
+    public final Long countUsers(String filter) throws UserStoreException {
+
```

Comment:
can we aggregate the functionality in countUsersClaims with this one? this is counting users with username claim, isn't it?

Reply:
+1. 

We have first-class support for usernames. Internally, we can identify the username based on the username claim URI. Then we can get rid of this method.